examples/leader-election/test.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+cleanup() {
+  exit_code=$?
+  set +e
+  echo "Rollback metacontroller..."
+  kubectl scale statefulset --replicas="$previous_replicas" -n metacontroller metacontroller
+  kubectl rollout undo statefulset metacontroller -n metacontroller
+  kubectl rollout status --watch --timeout=180s statefulset/metacontroller -n metacontroller
+  exit $exit_code
+}
+trap cleanup EXIT
+
+set -ex
+
+success_msg='successfully acquired lease'
+attempt_msg='attempting to acquire leader lease'
+
+previous_replicas=$(kubectl get statefulset metacontroller -n metacontroller -o=jsonpath='{.spec.replicas}')
+kubectl apply -k ./manifest
+kubectl scale statefulset --replicas=2 -n metacontroller metacontroller
+kubectl rollout status --watch --timeout=180s statefulset/metacontroller -n metacontroller
+
+# wait for one pod to acquire the leader lease
+until [[ "$(kubectl logs metacontroller-0 -n metacontroller | grep "$success_msg" | wc -l)" -eq 1 ||
+         "$(kubectl logs metacontroller-1 -n metacontroller | grep "$success_msg" | wc -l)" -eq 1 ]]; do
+  sleep 1
+done
+
+# determine which pods have attempted or successfully acquired the leader lease
+pod0_attempt=$(kubectl logs metacontroller-0 -n metacontroller | grep "$attempt_msg" | wc -l)
+pod0_success=$(kubectl logs metacontroller-0 -n metacontroller | grep "$success_msg" | wc -l)
+pod1_attempt=$(kubectl logs metacontroller-1 -n metacontroller | grep "$attempt_msg" | wc -l)
+pod1_success=$(kubectl logs metacontroller-1 -n metacontroller | grep "$success_msg" | wc -l)
+
+# both pods should attempt to acquire the leader lease
+if [[ "$pod0_attempt" -ne 1 || "$pod1_attempt" -ne 1 ]]; then
```

Comment:
query: Can we depend on this logic always? Perhaps a warning makes more sense.
query: Should it check for `-ne 0` instead of `-ne 1`

Reply:
You are right it could be the case that the first pod attempts and successfully acquires the lock before the second pod attempts. 👍